src/galaxy/SystemPath.cpp

```diff
@@ -27,9 +27,18 @@ static int ParseInt(const std::string &str)
 	int i = 0;
 	try {
 		i = std::stoi(str);
+#ifndef __GNUC__
+/* GCC doesnt know pragma warning push/disable/pop */
+#pragma message("FIX: warning of unused variable")
+#pragma warning(push)
+#pragma warning(disable : 4101)
+#endif /* __GNUC__ */
 	} catch (const std::invalid_argument &e) {
 		throw SystemPath::ParseFailure();
 	}
+#ifndef __GNUC__
+#pragma warning(pop)
+#endif /* __GNUC__ */
```

Comment:
Instead of the ugly preprocessor check to disable a warning number, please just remove the variable name or cast it to void; e.g. `catch (const std::invalid_argument &)`.

Reply:
Im a C++ noob, now I know how to